subprojects/plugins/plugins.gradle.kts

```diff
@@ -0,0 +1,77 @@
+import org.gradle.gradlebuild.unittestandcompile.ModuleType
+import org.gradle.gradlebuild.testing.integrationtests.cleanup.WhenNotEmpty
+
+/*
+ * Copyright 2010 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+dependencies {
+    compile(library("groovy"))
+
+    compile(project(":core"))
+    compile(project(":workers"))
+    compile(project(":dependencyManagement"))
+    compile(project(":reporting"))
+    compile(project(":platformJvm"))
+    compile(project(":languageJvm"))
+    compile(project(":languageJava"))
+    compile(project(":languageGroovy"))
+    compile(project(":diagnostics"))
+    compile(project(":testingJvm"))
+
+    compile(library("ant"))
+    compile(library("asm"))
+    compile(library("commons_io"))
+    compile(library("commons_lang"))
+    compile(library("slf4j_api"))
+
+    runtime(library("commons_cli"))
+
+    // This dependency makes the services provided by `:compositeBuilds` available at runtime for all integration tests in all subprojects
+    // Making this better would likely involve a separate `:gradleRuntime` module that brings in `:core`, `:dependencyManagement` and other key subprojects
+    runtime(project(":compositeBuilds"))
+
+    testFixturesApi(project(":internalIntegTesting"))
+
+    testCompile(testLibrary("jsoup"))
+
+    integTestRuntime(project(":maven"))
+}
+
+
+gradlebuildJava {
+    moduleType = ModuleType.WORKER
+}
+
+testFixtures {
+    from(":core")
+    from(":core", "testFixtures")
+    from(":dependencyManagement")
+    from(":languageJava")
+    from(":languageGroovy")
+    from(":diagnostics")
+}
+
+val wrapperJarDir = file("$buildDir/generated-resources/wrapper-jar")
+evaluationDependsOn(":wrapper")
+val wrapperJar by tasks.registering(Copy::class) {
+    from(project(":wrapper").tasks.named("executableJar"))
+    into(wrapperJarDir)
+}
+sourceSets[SourceSet.MAIN_SOURCE_SET_NAME].output.dir(wrapperJarDir, "builtBy" to wrapperJar)
```

Comment:
:x: `sourceSets.main { ... }`

Reply:
Why is `sourceSets.main { output... }` valid and `sourceSets.main.output...` giving me an error? I find that pretty counter-intuitive which is why I'm reverting to the `sourceSets["main"].output...` structure.

Reply:
First thing to keep in mind is that the Kotlin DSL syntactic sugar emphasis the lazy configuration API.

`sourceSets.main { output...}` is valid because `sourceSets.main` (the accessor, as we call it) returns a `NamedDomainObjectProvider<SourceSet>` and invoking a provider with a lambda like in `sourceSets.main { output... }` is equivalent to calling `configure` as in `sourceSets.main.configure { output...}`.

> I find that pretty counter-intuitive which is why I'm reverting to the sourceSets["main"].output... structure.

After the explanation I gave above, how do you feel about it?